outport/drivers/factory/elasticDriverFactory.go

```diff
@@ -25,70 +27,62 @@ type ArgsIndexerFactory struct {
 	Password                 string
 	Marshalizer              marshal.Marshalizer
 	Hasher                   hashing.Hasher
-	EpochStartNotifier       sharding.EpochStartEventNotifier
-	NodesCoordinator         sharding.NodesCoordinator
 	AddressPubkeyConverter   core.PubkeyConverter
 	ValidatorPubkeyConverter core.PubkeyConverter
 	TemplatesPath            string
-	Options                  *indexer.Options
+	Options                  *elastic.Options
 	EnabledIndexes           []string
 	Denomination             int
 	AccountsDB               state.AccountsAdapter
 	FeeConfig                *config.FeeSettings
 	IsInImportDBMode         bool
 }
 
-// NewIndexer will create a new instance of Indexer
-func NewIndexer(args *ArgsIndexerFactory) (indexer.Indexer, error) {
+// NewElasticClient will create a new instance of elastic client
+func NewElasticClient(args *ArgsElasticDriverFactory) (drivers.Driver, error) {
 	err := checkDataIndexerParams(args)
 	if err != nil {
 		return nil, err
 	}
 
-	if !args.Enabled {
-		return indexer.NewNilIndexer(), nil
```

Comment:
so we can remove the NilIndexer implementation.

Reply:
yes. NilIndexer is no needed anymore.